packages/cli/src/daemon/enable/Enable.handler.ts

```diff
@@ -0,0 +1,198 @@
+                    msg: `Unable to create directory '${zoweHomeBin}'.\nReason: ${err}`
+                });
+            }
+        }
+
+        // extract executable from the tar file into the bin directory
+        await this.unzipTgz(preBldTgz, zoweHomeBin, ImperativeConfig.instance.rootCommandName);
+
+        /* Even though we await the unzip above, the OS still considers the exe file in-use
+         * for a while. We will get the following error message when trying to run the exe.
+         * "The process cannot access the file because it is being used by another process."
+         * So, we wait a little bit.
+         */
+        const halfSecOfMillis = 500;
+        await CliUtils.sleep(halfSecOfMillis);
+
+        // display the version of the executable
+        let userInfoMsg: string = "Zowe CLI native executable version = ";
+        const zoweExePath = nodeJsPath.resolve(zoweHomeBin, "zowe");
+        const pipe: StdioOptions = ["pipe", "pipe", process.stderr];
+        try {
+            const execOutput = execSync(`"${zoweExePath}" --version-exe`, {
+                stdio: pipe
+            });
+            // remove any newlines from the version number
+            userInfoMsg += execOutput.toString().replace(/\r?\n|\r/g, "");
+        } catch (err) {
+            userInfoMsg += err.message;
+        }
+
+        // if ZOWE_CLI_HOME/bin is not on our PATH, add an instruction to add it
+        if (process.env?.PATH?.length > 0) {
+            if (!process.env.PATH.includes(zoweHomeBin)) {
+                userInfoMsg += `\n\nAdd '${zoweHomeBin}' to your path.` +
+                    "\nOtherwise, you will continue to run the classic Zowe CLI interpreter.";
+            }
+        }
+
+        // if ZOWE_USE_DAEMON is set, and turned off, add a warning message
+        if (process.env?.ZOWE_USE_DAEMON?.length > 0) {
+            switch (process.env.ZOWE_USE_DAEMON) {
+                case "no":
+                case "false":
+                case "0": {
+                    userInfoMsg += `\n\nYour ZOWE_USE_DAEMON environment variable is set to '${process.env.ZOWE_USE_DAEMON}'.` +
+                    "\nYou must remove it, or set it to 'yes' to use daemon mode.";
+                }
+            }
+        }
+
+        return userInfoMsg;
+    }
+
+    /**
+     * Unzip, from a gzipped tar file, any file that contains fileToExtract as a
+     * substring of the file name. The file will be placed into toDir.
+     * We expect toDir to already exist.
+     *
+     * @param tgzFile The gzipped tar file that we will extract
+     *
+     * @param toDir The directory into which we extract files
+     *
+     * @param fileToExtract The file name (or substring of the file name) to extract.
+     *
+     * @throws {ImperativeError}
+     * @returns A void promise to synchronize this operation.
+     */
+    private async unzipTgz(tgzFile: string, toDir: string, fileToExtract: string): Promise<void> {
```

Comment:
I do not object to this being a utility in another area, since it is a more general purpose utility. I kept it private, to remain consistent with our other decisions about private functions, and since I could not identify any other current need for this utility in other parts of our code. I would advocate moving this function when another piece of code needs to do the same thing.

Reply:
Agreed!
Let's wait until there is a use-case for exposing it 👍